src/index.js

```diff
@@ -96,8 +96,9 @@ function hypertext(render, postprocess) {
               // ignore
             } else if (value instanceof Node
                 || (typeof value !== "string" && value[Symbol.iterator])
-                || (/(?:^|>)$/.test(strings[j - 1]) && /^(?:<(?!\/textarea)|$)/.test(input))) {
+                || (/(?:^|>)$/.test(strings[j - 1]) && /^(?:<(?!\/(?:script|style|textarea|title))|$)/.test(input))) {
```

Comment:
I think instead of trying to lookahead at the following closing tag (which might lead to inconsistencies with the how we parse that closing tag, e.g., whitespace or case-sensitivity or mismatched closing tags), we’ll need to track which tag we’re inside when the tag opens. Specifically these places:

STATE_TAG_NAME ↦ STATE_DATA
https://github.com/observablehq/htl/blob/4038204f3e0946c75e46942c3167121852944066/src/index.js#L193

STATE_AFTER_ATTRIBUTE_NAME ↦ STATE_DATA
https://github.com/observablehq/htl/blob/4038204f3e0946c75e46942c3167121852944066/src/index.js#L229

STATE_BEFORE_ATTRIBUTE_NAME ↦ STATE_DATA
https://github.com/observablehq/htl/blob/4038204f3e0946c75e46942c3167121852944066/src/index.js#L244

STATE_ATTRIBUTE_VALUE_UNQUOTED ↦ STATE_DATA
https://github.com/observablehq/htl/blob/4038204f3e0946c75e46942c3167121852944066/src/index.js#L266

STATE_AFTER_ATTRIBUTE_VALUE_QUOTED ↦ STATE_DATA
https://github.com/observablehq/htl/blob/4038204f3e0946c75e46942c3167121852944066/src/index.js#L276

In addition, we’ll need to update the `nameStart` and `nameEnd` indexes to track the current tag name, similar to how we track the current attribute name (and clear them when exiting the tag, and possibly use separate `attrNameStart` and `tagNameStart` local variables for clarity), e.g.:

https://github.com/observablehq/htl/blob/4038204f3e0946c75e46942c3167121852944066/src/index.js#L204

If we do this, we should also be able to fix #6.

Reply:
For inspiration, take a look at the Julia fork of this project which has addressed this issue:

https://github.com/clarkevans/HypertextLiteral.jl/blob/c33b88ad288e0061ee3c8b919a83c5689a9929e2/src/lexer.jl#L37-L45